cmd/config.go

```diff
@@ -177,6 +180,71 @@ func readEnvConfig() (conf Config, err error) {
 	return conf, nil
 }
 
+// This checks for conflicting options and turns any shortcut options (i.e. duration, iterations,
+// stages) into the proper scheduler configuration
+func buildExecutionConfig(conf Config) (Config, error) {
+	result := conf
+	if conf.Duration.Valid {
+		if conf.Iterations.Valid {
+			//TODO: make this an error in the next version
+			log.Warnf("Specifying both duration and iterations is deprecated and won't be supported in the future k6 versions")
+		}
+
+		if conf.Stages != nil {
+			//TODO: make this an error in the next version
+			log.Warnf("Specifying both duration and stages is deprecated and won't be supported in the future k6 versions")
+		}
+
+		if conf.Execution != nil {
+			return result, errors.New("specifying both duration and execution is not supported")
```

Comment:
I would like to actually add different type for configuration errors. Some of the common ones can also be predefined as `const`  if possible so that we can check for them where needed :)